### PR TITLE
Feat/search page UI upgrade

### LINKED ISSUE #59: Enhance the searching for locations in Search Page
After integrating Google Places API, enhance the search page to use the locations and allow filtering for locations.

Image

### PR DESCRIPTION
**Pull request by apana03**
## Summary

Refined the Search Page to look like the Mockup provided in the figma.

## Changes
- **Screens/UI:** 2 tabs are now present (Users and Places), also when in places tab we can toggle between map view and list view, also changed what composables are show for each user and each place.
- **Repositories:** Added lat_lng Place field in the Places API request to be able to display a pin in the map view for each place.

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #59 ` (if applicable).
- [x] Tests have been added for new functionality.
- [x] Screenshots or UI changes have been attached.

##  Testing
- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

##  Related Issues/Tickets

Closes #59

## Screenshots (if applicable)
image

imageWON

image

### CODE REVIEW

**Comment by g0g05arui** [comment]
@apana03 Please fix the CI pipeline as well, need to remove the quotes from the secret in github.

**Comment by apana03** [comment]
@g0g05arui Done, everything works as expected now. Take a look and tell me if you see anything else.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
         env:
           PLACES_API_KEY: ${{ secrets.PLACES_API_KEY }}
           MAPS_API_KEY: $ {{ secrets.PLACES_API_KEY }}
-        run: echo PLACES_API_KEY=\"$PLACES_API_KEY\" > ../../local.properties
+        run: echo PLACES_API_KEY=\"$PLACES_API_KEY\" > ./local.properties
```
Comment (g0g05arui) [inline .github/workflows/build.yaml:60]: Please remove the quotes

**Review by mirceagavrizi — APPROVED** [review]
Otherwise it looks good, nice job with the tests.

**Pull request merged**